x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -90,11 +93,6 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
                         e -> handleException("Secondary authentication", request, channel, e)));
                 }, e -> handleException("Authentication", request, channel, e)));
         } else {
-            if (request.method() != Method.OPTIONS) {
-                HeaderWarning.addWarning("Elasticsearch built-in security features are not enabled. Without authentication, your cluster " +
-                    "could be accessible to anyone. See https://www.elastic.co/guide/en/elasticsearch/reference/" + Version.CURRENT.major +
-                    "." + Version.CURRENT.minor + "/security-minimal-setup.html to enable security.");
```

Comment:
I have discussed it with Fabio and he agreed that with Security On by Default we won't need it in 8.0. The change which is making the default behaviour to true is on it's way.

Reply:
As discussed, I am happy to let it pass if Tim is OK with the approach. Thanks!

Reply:
I would prefer that we kept the warning. Removing features based on the premise that another change will come along and make them redundant is a risky business. If we change our plans for security on by default, then this warning might still be required, but it will have been lost.  
The likelihood is relatively low, but the safer option is to remove code when it's no longer needed rather than removing it when we decide it won't be needed in the future.

However, in this case, there's no simple way to keep the code around.
We need to know whether security is _implictly disabled_, and there's no method on `XPackLicenseState` for that, and I don't want to add one now.  
We can looking at `XPackSettings.SECURITY_ENABLED`, but then we need a copy of the node's `Settings`, which we don't have.

Because it's a warning, and not critical functionality, I think it's easier to just remove it, but that's not my preferred approach in general.